internal/policygen/iam.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policygen
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/terraform"
+	"github.com/hashicorp/terraform/states"
+)
+
+type binding struct {
+	Type   string
+	ID     string
+	Role   string
+	Member string
+}
+
+func generateIAMBindingsPolicies(resources []*states.Resource, outputDir string) error {
+
+	bindings, err := bindings(resources)
+	if err != nil {
+		return err
+	}
+	log.Printf("Found %d bindings from input Terraform resources", len(bindings))
+	// TODO(xingao): Support google_*_iam_bindings.
+	// TODO(xingao): Generate policies.
+
+	return nil
+}
+
+func bindings(resources []*states.Resource) ([]*binding, error) {
+	typeToIDField := map[string]string{
+		"project":      "project",
+		"folder":       "folder",
+		"organization": "org_id",
+	}
+
+	var bindings []*binding
+	for t, idField := range typeToIDField {
+		instances, err := terraform.GetInstancesForType(resources, fmt.Sprintf("google_%s_iam_member", t))
+		if err != nil {
+			return nil, fmt.Errorf("get resource instances for type %q: %v", fmt.Sprintf("google_%s_iam_member", t), err)
```

Comment:
Nit: This is reused in two places, should just store it in a variable.

Reply:
Done